fillers/eips/eip4844/datahash_opcode.py

```diff
@@ -538,118 +545,238 @@ def test_datahash_gas(_: str):
+        ]
     )
-    datahash_max_gas_measure_code_address = to_address(0x200)
+    gas_measures_code = [
+        CodeGasMeasure(
+            code=getattr(Op, "PUSH32")(gas_measure) + Op.DATAHASH,
+            overhead_cost=3,
+            extra_stack_items=1,
+        )
+        for gas_measure in gas_measures
+    ]
 
-    # Store all contracts in pre-state
     pre = {
         TestAddress: Account(balance=1000000000000000000000),
-        datahash_zero_gas_measure_code_address: Account(
-            code=datahash_zero_gas_measure_code
-        ),
-        datahash_max_gas_measure_code_address: Account(
-            code=datahash_max_gas_measure_code
-        ),
     }
+    post = {}
+
+    def create_tx(ty: int, to: str, nonce: int):
+        return Transaction(
+            ty=ty,
+            nonce=nonce,
+            data=to_hash_bytes(0),
+            to=to,
+            gas_limit=3000000,
+            max_fee_per_gas=10,
+            max_priority_fee_per_gas=10,
+            max_fee_per_data_gas=10,
+            access_list=[],
+            blob_versioned_hashes=[to_hash_bytes(2**256 - 1)]
+            if ty == 5
+            else None,
+        )
+
+    txs_ty2, txs_ty5 = [], []
+    for i, code_gas_measure in enumerate(gas_measures_code):
+        address = to_address(0x100 + i * 0x100)
+        pre[address] = Account(code=code_gas_measure)
+        txs_ty2.append(create_tx(ty=2, to=address, nonce=i))
+        txs_ty5.append(create_tx(ty=5, to=address, nonce=i))
+        post[address] = Account(storage={0: DATAHASH_GAS_COST})
+
+    # DATAHASH gas cost on tx type 2
+    yield StateTest(
+        env=env,
+        pre=pre,
+        post=post,
+        txs=txs_ty2,
+        tag="tx_type_2",
+    )
 
     # DATAHASH gas cost on tx type 5
-    tx1 = Transaction(
-        ty=5,
-        nonce=0,
-        data=to_hash_bytes(0),
-        to=datahash_zero_gas_measure_code_address,
-        gas_limit=3000000,
-        max_fee_per_gas=10,
-        max_priority_fee_per_gas=10,
-        max_fee_per_data_gas=10,
-        access_list=[],
-        blob_versioned_hashes=[to_hash_bytes(2**256 - 1)],
+    # Can only use 4 txs (max blobs) per block
+    num_blocks = (len(txs_ty5) + MAX_BLOB_PER_BLOCK - 1) // MAX_BLOB_PER_BLOCK
+    blocks = [
+        Block(
+            txs=txs_ty5[i * MAX_BLOB_PER_BLOCK : (i + 1) * MAX_BLOB_PER_BLOCK]
+        )
+        for i in range(num_blocks)
+    ]
+
+    yield BlockchainTest(
+        pre=pre,
+        post=post,
+        blocks=blocks,
+        tag="tx_type_5",
     )
-    tx2 = Transaction(
-        ty=5,
-        nonce=1,
-        data=to_hash_bytes(0),
-        to=datahash_max_gas_measure_code_address,
-        gas_limit=3000000,
-        max_fee_per_gas=10,
-        max_priority_fee_per_gas=10,
-        max_fee_per_data_gas=10,
-        access_list=[],
-        blob_versioned_hashes=[to_hash_bytes(2**256 - 1)],
+
+
+@test_from(fork=ShardingFork)
+def test_datahash_blob_versioned_hash(_: str):
```

Comment:
We could add another test case where we request `DATAHASH` multiple times, to check if the result is the same.
E.g.:
```
- DATAHASH(0x00), DATAHASH(0x00)
- DATAHASH(0x00), DATAHASH(0xff..ff), DATAHASH(0x00)
- DATAHASH(0x00), DATAHASH(0x01), DATAHASH(0x00)
```
Where blobs 0x00 and 0x01 exist, just to name a few.